### PR TITLE
[MB-1552] [AMQP] Provide support for prefixed wildcards as per specification

### PR DESCRIPTION
**Pull request by a5anka**
We were checking if a destination contains ".*" or ".#" to identiify
a wildcard destination in AMQPUtils#isWildCardDestination.
Removed "." character infront of wild card character to identify
wildcard prefixed destinations (such as *.SriLanka)

https://wso2.org/jira/browse/MB-1552

### CODE REVIEW

**Comment by a5anka** [comment]
This might conflict with @akalankapagoda 's changes

File: modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/ClusterSubscriptionBitMapHandler.java [hunk 1]
```diff
@@ -238,25 +238,25 @@ private void addSubscriptionColumn(String subscribedDestination, int subscriptio
...
-                        bitSet.set(subscriptionIndex, isMatchForSubscriptionType(wildcardDestination,
-                                matchDestinationForNull));
-//                    }
+                    // Check if this constituent being null matches the destination if we match it with
+                    // a null constituent
+                    String wildcardDestination = NULL_CONSTITUENT + constituentsDelimiter +
+                                                 currentConstituent;
+                    bitSet.set(subscriptionIndex, isMatchForSubscriptionType(wildcardDestination,
+                                                                             matchDestinationForNull));
+                    //                    }
```
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/ClusterSubscriptionBitMapHandler.java]: Shall we remove commented code?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/ClusterSubscriptionBitMapHandler.java [hunk 2]
```diff
@@ -238,25 +238,25 @@ private void addSubscriptionColumn(String subscribedDestination, int subscriptio
...
+                if (constituentOfCurrentRow.equals(currentConstituent)) {
                     bitSet.set(subscriptionIndex);
                 } else if (NULL_CONSTITUENT.equals(constituentOfCurrentRow)) {
-                        // Check if this constituent being null matches the destination if we match it with
-                        // a null constituent
-                        String wildcardDestination = NULL_CONSTITUENT + constituentsDelimiter +
-                                currentConstituent;
-                        bitSet.set(subscriptionIndex, isMatchForSubscriptionType(wildcardDestination,
-                                matchDestinationForNull));
-//                    }
```
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/ClusterSubscriptionBitMapHandler.java]: Shall we remove commented code?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/ClusterSubscriptionBitMapHandler.java (hunk 1 above)
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/ClusterSubscriptionBitMapHandler.java]: :+1:

**Comment by akalankapagoda** [comment]
Will resolve conflicts and push as a new pull request.

**Pull request closed without merging**